torch/nn/modules/loss.py

```diff
@@ -1235,24 +1240,45 @@ class CTCLoss(_Loss):
 
     Inputs:
         log_probs: Tensor of size :math:`(T, N, C)` where `C = number of characters in alphabet including blank`,
-            `T = input length`, and `N = batch size`.
+        `T = input length`, and `N = batch size`.
             The logarithmized probabilities of the outputs
             (e.g. obtained with :func:`torch.nn.functional.log_softmax`).
-        targets: Tensor of size :math:`(N, S)` or `(sum(target_lengths))`.
-            Targets (cannot be blank). In the second form, the targets are assumed to be concatenated.
+        targets: Tensor of size :math:`(N, S)` or `(sum(target_lengths))`, where `N = batch size`, and
+        `S = maximum target length`.
+            Each element in the target sequence is a class index. Target index cannot be blank (default=0).
+            In the second form, the targets are assumed to be concatenated.
         input_lengths: Tuple or tensor of size :math:`(N)`.
-            Lengths of the inputs (must each be :math:`\leq T`)
+            Lengths of the inputs (must each be :math:`\leq T`). Lengths are specified for each sequence to achieve
+            masking. In effect, length is the stop index s for each target sequence,
+            such that :math: `target_n = targets[n,:s_n]` for each target in a batch.
         target_lengths: Tuple or tensor of size  :math:`(N)`.
-            Lengths of the targets
+            Lengths of the targets (may be :math: < S)
 
     Example::
 
+        >>> T = 50  # Input sequence length
+        >>> C = 20  # Number of classes (excluding blank)
+        >>> N = 16  # Batch size
+        >>> S = 30  # Target sequence length of longest target in batch
+        >>>
+        >>> min_target_length = 10
+        >>> max_target_length = S
+        >>>
```

Comment:
I must admit that the blank lines are just too much. Also, I wonder, if we need to introduce a second name for S.

Reply:
I redid the naming it in the latest update. Also removed some of the self explanatory comments, and removed some spaces.